Breeze/src/Breeze.Wallet/LightWalletSyncManager.cs

```diff
@@ -157,19 +129,66 @@ public void ProcessBlock(Block block)
-            this.SyncFrom(blockSyncStart);
+            // before we start syncing we need to make sure that the chain is at a certain level.
+            // if the chain is behind the date from which we want to sync, we wait for it to catch up, and then we start syncing.
+            // if the chain is already past the date we want to sync from, we don't wait, even though the chain might not be fully downloaded.
+            if (this.chain.Tip.Header.BlockTime.LocalDateTime < date)
+            {
+                AsyncLoop.RunUntil("WalletFeature.DownloadChain", this.cancellationProvider.Cancellation.Token,
+                    () => this.chain.Tip.Header.BlockTime.LocalDateTime >= date,
+                    () => this.StartSync(this.chain.GetHeightAtTime(date)),
+                        (ex) =>
+                        {
+                            // in case of an exception while waiting for the chain to be at a certain height, we just cut our losses and 
+                            // sync from the current height.
+                            this.logger.LogError($"Exception occurred while waiting for chain to download: {ex.Message}");
+                            this.StartSync(this.chain.Tip.Height);
+                        },
+                    TimeSpans.FiveSeconds);
+            }
+            else
+            {               
+                this.StartSync(this.chain.GetHeightAtTime(date));
+            }
         }
 
+        /// <inheritdoc />
         public void SyncFrom(int height)
         {
+            // before we start syncing we need to make sure that the chain is at a certain level.
+            // if the chain is behind the height from which we want to sync, we wait for it to catch up, and then we start syncing.
+            // if the chain is already past the height we want to sync from, we don't wait, even though the chain might  not be fully downloaded.
+            if (this.chain.Tip.Height < height)
+            {
+                AsyncLoop.RunUntil("WalletFeature.DownloadChain", this.cancellationProvider.Cancellation.Token,
+                    () => this.chain.Tip.Height >= height,
+                    () => this.StartSync(height),
+                    (ex) =>
+                    {
+                        // in case of an exception while waiting for the chain to be at a certain height, we just cut our losses and 
+                        // sync from the current height.
+                        this.logger.LogError($"Exception occurred while waiting for chain to download: {ex.Message}");
+                        this.StartSync(this.chain.Tip.Height);
+                    },
+                    TimeSpans.FiveSeconds);
+            }
+            else
+            {
+                this.StartSync(height);
+            }
+        }
+
+        private void StartSync(int height)
+        {
             var chainedBlock = this.chain.GetBlock(height);
             if (chainedBlock == null)
```

Comment:
Because this might be called on Initialize we need to replicate the recover code here as well.
```
                  if (this.walletTip == null && this.chain.Height > 0)
                  {
                      // recover
                  }
```
This can be a TODO not urgent for now.